go-controller/pkg/allocator/mac/reservation_test.go

```diff
@@ -0,0 +1,75 @@
+package mac_test
+
+import (
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/allocator/mac"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("ReservationManager", func() {
+	const owner1 = "namespace1/pod1"
+	const owner2 = "namespace2/pod2"
+
+	var testMgr *mac.ReservationManager
+	var mac1 net.HardwareAddr
+
+	BeforeEach(func() {
+		var err error
+		mac1, err = net.ParseMAC("aa:bb:cc:dd:ee:f1")
+		Expect(err).NotTo(HaveOccurred())
+
+		testMgr = mac.NewManager()
+	})
+
+	Context("reserve", func() {
+		DescribeTable("should not reserve MAC, given",
+			func(owner string, mac net.HardwareAddr) {
+				Expect(testMgr.Reserve(owner, mac)).To(Succeed())
+
+				Expect(testMgr.Reserve(owner1, mac1)).To(Succeed(), "mac should not be reserved")
+			},
+			Entry("zero-values", "", nil),
+			Entry("no owner", "", mac1),
+			Entry("no mac", owner1, nil),
+		)
+		It("should not fail on repeated reservation for the same owner", func() {
+			Expect(testMgr.Reserve(owner1, mac1)).To(Succeed())
+			Expect(testMgr.Reserve(owner1, mac1)).To(Succeed(), "same owner should not raise a conflict")
+
+			Expect(testMgr.Reserve(owner2, mac1)).To(MatchError(mac.ErrMACConflict),
+				"different owner should raise a conflict")
+		})
+	})
+
+	Context("release a reserved mac", func() {
+		BeforeEach(func() {
+			By("reserve mac1 for owner1")
+			Expect(testMgr.Reserve(owner1, mac1)).To(Succeed())
+		})
+
+		DescribeTable("should not release MAC, given",
+			func(owner string, macAddr net.HardwareAddr) {
+				testMgr.Release(owner, macAddr)
+
+				Expect(testMgr.Reserve(owner2, mac1)).To(MatchError(mac.ErrMACConflict),
+					"mac1 reserved for owner1, it should raise a conflict")
+			},
+			Entry("zero-values", "", nil),
+			Entry("no owner", "", mac1),
+			Entry("no mac", owner1, nil),
+		)
+
+		It("should release MAC", func() {
+			Expect(testMgr.Reserve(owner2, mac1)).To(MatchError(mac.ErrMACConflict),
+				"fail because mac1 reserved for owner1")
+
+			testMgr.Release(owner1, mac1)
+
+			Expect(testMgr.Reserve(owner2, mac1)).To(Succeed(),
+				"reserved mac1 for owner2 because its has been released")
+		})
```

Comment:
can you add a scenario where a MAC is reserved, and a different owner tried to release? 
We should document what is the wanted behavior.

Reply:
Done, added to the DescribeTable at line 54